### ISSUE TITLE
[Research Topic] Mask Topology Simplification & MBMW-Aware Post-Processing

### ISSUE THREAD

**Issue by telleroutlook** [OpenLithoHub/OpenLithoHub#9]
# Motivation

OpenLithoHub's curvilinear ILT path (`LevelSetILTModel`, `NeuralILTModel`) produces beautifully optimised masks — and routinely produces masks that the **multi-beam mask writer (MBMW)** in the mask shop cannot faithfully print. Sub-spec hairs, sliver islands, sub-tip-radius corners, and 1–2 px bridges all sail through the optimiser because the loss only cares about wafer-side fidelity. Today these defects are caught post-hoc by `src/openlithohub/benchmark/compliance/mrc.py` — too late for the optimization loop, and the resulting GDS may still be unprintable.

We need a **mask-topology simplifier** that sits between curvilinear ILT and GDS export, and that is good enough that:

1. The aerial-image / EPE / PV-band metrics of the simplified mask are within tolerance of the optimised mask.
2. The simplified mask passes a fixed MBMW rule set (min CD, min spacing, max curvature κ, no sliver islands).
3. Contours come out as smooth B-splines, not pixel-staircased polygons.

This is complementary to:
- **#8** (differentiable MRC loss) — *prevents* MRC violations during optimisation. This issue is about *cleaning up* whatever the optimiser still emits.
- **#6** (generative SRAF) — covers SRAF *placement*. This issue covers SRAF + main-pattern *post-processing*.

# What we want

A new module — sketch path: `src/openlithohub/workflow/contour/mbmw_filter.py` (or, if the approach is learned, `src/openlithohub/models/mask_simplifier.py`):

```python
import torch

def simplify_for_mbmw(
    mask: torch.Tensor,            # (H, W) continuous in [0, 1] or binary
    *,
    min_cd_px: int = 4,
    min_spacing_px: int = 4,
    max_curvature_per_px: float = 0.25,
    metric_tolerance_nm: float = 1.0,
) -> torch.Tensor:
    """Return an MBMW-printable mask whose aerial-image response is within
    ``metric_tolerance_nm`` of the input mask.

    Hairs, slivers, and sub-tip-radius corners are removed; remaining
    contours are smoothed to a curvature bounded by ``max_curvature_per_px``.
    """
```

Two implementation routes are both very welcome:

- **Geometric** — distance-transform island merging + B-spline contour fitting + curvature-aware smoothing (Taubin-style). Uses utilities already in `src/openlithohub/_utils/morphology.py` and `src/openlithohub/workflow/contour/curvilinear.py`.
- **Learned** — a small autoencoder denoiser trained on (raw curvilinear, MBMW-cleaned) pairs. Bonus: the latent space could double as a curvilinear-mask compression scheme.

# Interface contract

> **No subclassing required.** Just write the function above. EPE / PV-band / MRC re-evaluation after simplification, GDS export, and integration into the leaderboard pipeline are handled by maintainers. If you take the learned route, the `LithographyModel` interface is the natural home — see `src/openlithohub/models/base.py`.

# Suggested deliverables

1. The function (or model) above, with tests against a handful of `LevelSetILTModel` outputs.
2. A short notebook showing **before/after** — aerial-image SSIM, EPE, PV-band, and MRC pass-rate before vs after simplification. The promise is "MRC pass-rate goes from 30 % to 95 %, EPE drift < 1 nm."
3. (Stretch) A learned baseline alongside the geometric one, with a fair head-to-head on the same set of curvilinear outputs.

# Why this is a great research target

- **An open hole in OSS lithography** — academic ILT papers stop at the mask; nobody publishes on MBMW-aware post-processing.
- **Sits at the geometry / DL boundary** — equally publishable as a CG paper (curvature-bounded simplification) and a DL paper (mask denoiser).
- **Conferences that love this topic**: SPIE Photomask Technology, SPIE Advanced Lithography, ICCAD, DAC, EuroGraphics (geometric route).

# Recommended starting points

- Taubin, G. (1995), "Curve and surface smoothing without shrinkage" — for the curvature-bounded smoothing primitive.
- Pang, L. et al., recent ILT-MBMW co-optimisation work in SPIE Photomask 2023 / 2024 proceedings.
- For the learned route: any small mask-denoising autoencoder; start from the encoder of `src/openlithohub/models/layout_mae.py`.

# Win-win promise

> *This is a highly open research problem perfect for PhD students or ML researchers looking to publish at EDA / lithography conferences (SPIE, ICCAD, DAC). If you decide to tackle this, the core maintainers will provide full support on the OpenLithoHub framework side. We encourage you to publish papers based on this, and we will feature your merged implementation and paper on our official website's [Algorithms Showcase](https://openlithohub.com/showcase) section.*

---

Labels: `research-topic`, `phd-friendly`, `help wanted`, `call-for-paper`
Cross-refs: #6, #8
